beacon-chain/core/electra/consolidations.go

```diff
@@ -182,22 +189,44 @@ func ProcessConsolidationRequests(ctx context.Context, st state.BeaconState, req
 		if ctx.Err() != nil {
 			return fmt.Errorf("cannot process consolidation requests: %w", ctx.Err())
 		}
+		if IsValidSwitchToCompoundingRequest(st, cr) {
+			srcIdx, ok := st.ValidatorIndexByPubkey(bytesutil.ToBytes48(cr.SourcePubkey))
+			if !ok {
+				log.Error("failed to find source validator index")
+				continue
+			}
+			if err := SwitchToCompoundingValidator(st, srcIdx); err != nil {
+				log.WithError(err).Error("failed to switch to compounding validator")
+			}
+			continue
+		}
+		sourcePubkey := bytesutil.ToBytes48(cr.SourcePubkey)
+		targetPubkey := bytesutil.ToBytes48(cr.TargetPubkey)
+		if sourcePubkey == targetPubkey {
```

Comment:
+1, bytes.Equal does not alloc, but converting to [48]byte is a 48 byte alloc

Reply:
We would still need `[48]byte` when calling `ValidatorIndexByPubkey` in L224
```
srcIdx, ok := st.ValidatorIndexByPubkey(sourcePubkey)
tgtIdx, ok := st.ValidatorIndexByPubkey(targetPubkey)
```